pkg/controllers/user/ingresshostgen/ingresshostgen.go

```diff
@@ -0,0 +1,67 @@
+package ingresshostgen
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/rancher/pkg/settings"
+	v1beta12 "github.com/rancher/types/apis/extensions/v1beta1"
+	"github.com/rancher/types/config"
+	"k8s.io/api/extensions/v1beta1"
+)
+
+type IngressHostGen struct {
+	ingress v1beta12.IngressInterface
+}
+
+func Register(userOnlyContext *config.UserOnlyContext) {
+	c := &IngressHostGen{
+		ingress: userOnlyContext.Extensions.Ingresses(""),
+	}
+	userOnlyContext.Extensions.Ingresses("").AddHandler("ingress-host-gen", c.sync)
+}
+
+func (i *IngressHostGen) sync(key string, obj *v1beta1.Ingress) error {
+	if obj == nil {
+		return nil
+	}
+
+	ipDomain := settings.IngressIPDomain.Get()
+	if ipDomain == "" {
+		return nil
+	}
+
+	var xipHost string
+	for _, status := range obj.Status.LoadBalancer.Ingress {
+		if status.IP != "" {
+			xipHost = fmt.Sprintf("%s.%s.%s.%s", obj.Name, obj.Namespace, status.IP, ipDomain)
```

Comment:
Load Balancer ingress can also have a HostName, do we need to have an extra check for that?

Reply:
@alena1108 We can't support hostname.  We can't do a CNAME. So this only works for IP and if we don't find an IP we don't modify the ingress Host